File: HospitalManagement/src/app/models/appointment-dto.ts
export enum Gender {
  Male = 'Male',
  Female = 'Female',
  Other = 'Other',
}

export enum AppointmentStatus {
  Scheduled = 'Scheduled',
  Arrived = 'Arrived',
  // InConsultation = 'InConsultation',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
  NoShow = 'NoShow',
}

export enum AppointmentType {
  General = 'General',
  FollowUp = 'FollowUp',
  Emergency = 'Emergency',
  PostOperative = 'PostOperative',
  Preventive = 'Preventive',
  Referral = 'Referral',
}
export enum VisitType {
  FirstVisit = 'FirstVisit',
  FollowUp = 'FollowUp',
  TestReview = 'TestReview',
  Others = 'Others',
}
export interface Appointment {
  appointmentId: number;
  patientId: number;
  //newly add
  firstName: string;
  lastName: string;
  visitType: VisitType;
  gender: Gender;
  //for Doctor
  docFirstName?: string;
  docLastName?: string;
  doctorDepartment?: string;

  doctorId: number;
  admissionId?: number;
  tokenId?: number;
  appointmentDate: Date;
  patientPhone?: string;
  referralCode?: string;
  appointmentStatus: AppointmentStatus;
  appointmentType: AppointmentType;
}

export interface AppointmentDTO {
  appointmentId?: number;
  patientNo: string;
  firstName: string;
  lastName: string;
  age: number;
  gender: Gender;
  patientId?: number;
  doctorId: number;
  admissionId?: number | null;
  tokenId?: number | null;
  appointmentDate: Date | string; // Can handle both Date object and ISO string
  patientPhone?: string | null;
  referralCode?: string | null;
  appointmentStatus: AppointmentStatus;
  appointmentType: AppointmentType;
  visitType: VisitType;
}

File: HMSProjectOfMine/HMSProjectOfMine/DTOs/AppointmentDTO.cs
using HMSProjectOfMine.Enums;
using HMSProjectOfMine.Models;

namespace HMSProjectOfMine.DTOs
{
    public class AppointmentDTO
    {
        public int AppointmentId { get; set; }

        public string PatientNo { get; set; } = null!; // ✅ Required for identifying patient
                                                       //for patient:
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public VisitType VisitType { get; set; }

        //for Doctor Name:

        public string? DocFirstName { get; set; } = null!;
        public string? DocLastName { get; set; } = null!;
        public string? DoctorDepartment { get; set; }

        public int Age { get; set; }
        public Gender Gender { get; set; }
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public int? AdmissionId { get; set; }
        public int? TokenId { get; set; }

        // Appointment Details
        public DateTime AppointmentDate { get; set; }
        public string? PatientPhone { get; set; }
        public string? ReferralCode { get; set; }

        // Status and Type (Enums)
        public AppointmentStatus AppointmentStatus { get; set; } = AppointmentStatus.Scheduled;
        public AppointmentType AppointmentType { get; set; } = AppointmentType.General;
    }
}
